olive/systems/azureml/aml_system.py

```diff
@@ -21,13 +21,23 @@
 from olive.constants import Framework
 from olive.evaluator.metric import Metric, MetricResult
 from olive.hardware.accelerator import AcceleratorSpec
-from olive.model import ModelConfig, ModelStorageKind, OliveModel, ONNXModel
+from olive.model import ModelConfig, OliveModel
 from olive.passes.olive_pass import Pass
+from olive.resource_path import AZUREML_RESOURCE_TYPES, AzureMLModelConfig, ResourcePath, ResourceType
 from olive.systems.common import AzureMLDockerConfig, SystemType
 from olive.systems.olive_system import OliveSystem
 
 logger = logging.getLogger(__name__)
 
+RESOURCE_TYPE_TO_ASSET_TYPE = {
```

Comment:
How about maintain this map under `resource_type.py`?

Reply:
I kept this here since the azureml packages are optional and we cannot have this mapping in the `resource_paths.py` file. It is also only relevant to the azureml system so I think it is better for the azureml system to maintain this mapping?

Reply:
Asking this is because, we clone all ResourceType and set the mapped aml asset types. When there is any other similar requests, we may need to maintain that again in other system.

But currently it looks ok, as only aml is using that and we need to resolve the issue of packaging dependency. Thanks.

Reply:
Sounds good to me. 

Originally, I was thinking about putting the responsibility of creating aml input and args on the resource_paths so that we can add more resource types without having to touch the aml system. Maybe we can see if that's a better idea in the future?

Reply:
Yes. Sounds good.